week6/unique0902/src/ProductTable.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import ProductCategoryRow from './ProductCategoryRow';
+import ProductRow from './ProductRow';
+
+const ProductTable = ({
+  products,
+  filterText,
+  inStockOnly,
+  deleteProduct,
+  editProduct,
+}) => {
+  const rows = [];
+  let lastCategory = null;
+  let nowTimeKey = Date.now();
+  const productsCopy = [...products];
+  productsCopy
+    .sort((a, b) => (a.category > b.category ? 1 : -1))
+    .map((product) => {
+      if (product.name.toLowerCase().indexOf(filterText.toLowerCase()) === -1) {
+        return;
+      }
+
+      if (inStockOnly && !product.stocked) {
+        return;
+      }
+
+      if (product.category !== lastCategory) {
+        rows.push(
+          <ProductCategoryRow
+            category={product.category}
+            key={product.category}
+          />
+        );
+      }
+
+      rows.push(
+        <ProductRow
+          product={product}
+          key={nowTimeKey}
+          deleteProduct={deleteProduct}
+          editProduct={editProduct}
+        />
+      );
+      nowTimeKey++;
+      lastCategory = product.category;
+    });
+
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Price</th>
+        </tr>
+      </thead>
+      <tbody>{rows}</tbody>
```

Comment:
```suggestion
      <tbody>
        {filteredProducts.map((product, index) => {
          const categoryRow = product.category !== lastCategory && (
            <ProductCategoryRow
              category={product.category}
              key={product.category}
            />
          );

          lastCategory = product.category;

          return (
            <React.Fragment key={index}>
              {categoryRow}
              <ProductRow
                product={product}
                deleteProduct={deleteProduct}
                editProduct={editProduct}
              />
            </React.Fragment>
          );
        })}
      </tbody>
```

코드 보시고 궁금한거 있으시면 여기서 멘션하시거나 디스코드 웹 잡담방에 질문 올려주세요!..

Reply:
관련하여 디스코드 웹 질문방에 질문올렸습니다!